src/components/ContactsList/ContactRow.jsx

```diff
@@ -0,0 +1,138 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+const contactPropTypes = {
+  phone: PropTypes.shape({
+    number: PropTypes.string.isRequired,
+    type: PropTypes.string,
+    label: PropTypes.string,
+    primary: PropTypes.bool
+  }),
+  email: PropTypes.shape({
+    address: PropTypes.string.isRequired,
+    primary: PropTypes.bool,
+    type: PropTypes.string,
+    label: PropTypes.string
+  }),
+  name: PropTypes.shape({
+    familyName: PropTypes.string,
+    givenName: PropTypes.string,
+    additionalName: PropTypes.string,
+    namePrefix: PropTypes.string,
+    nameSuffix: PropTypes.string
+  })
+};
+
+const btohex = (name = "") => {
+  const colors = [
+    "#1abc9c",
+    "#2ecc71",
+    "#3498db",
+    "#9b59b6",
+    "#34495e",
+    "#16a085",
+    "#27ae60",
+    "#2980b9",
+    "#8e44ad",
+    "#2c3e50",
+    "#f1c40f",
+    "#e67e22",
+    "#e74c3c",
+    "#95a5a6",
+    "#f39c12",
+    "#d35400",
+    "#c0392b",
+    "#bdc3c7",
+    "#7f8c8d"
+  ];
+  return colors[
+    Array.from(name.toUpperCase())
+      .map(letter => letter.charCodeAt(0))
+      .reduce((sum, number) => sum + number, 0) % colors.length
+  ];
+};
+
+const ContactBadge = ({ name }) => {
+  const style = {
+    backgroundColor: `${btohex(name.givenName)}`,
+    color: "white"
+  };
+  return (
+    <div className="contact-badge" style={style}>
+      <span className="contact-badge-name">
+        {[name.givenName[0], name.familyName[0]].join("")}
```

Comment:
A contact doesn't always have  the `givenName` or `familyName` prop !

Reply:
I missigned this component: it should not await for a `name` object with specific property, it should accept a `firstname` and a `lastname` (I will see later for company names, etc. because the io.cozy.contacts specs are not complete).